integration-cli/docker_cli_search_test.go

```diff
@@ -16,10 +16,20 @@ func (s *DockerSuite) TestSearchOnCentralRegistry(c *check.C) {
 }
 
 func (s *DockerSuite) TestSearchStarsOptionWithWrongParameter(c *check.C) {
-	out, _, err := dockerCmdWithError("search", "--stars=a", "busybox")
+	out, _, err := dockerCmdWithError("search", "--filter", "has", "-stars", "=a", "busybox")
 	c.Assert(err, check.NotNil, check.Commentf(out))
 	c.Assert(out, checker.Contains, "invalid value", check.Commentf("couldn't find the invalid value warning"))
 
+	// -s --stars deprecated since Docker 1.13
+	out, _, err = dockerCmdWithError("search", "--stars=a", "busybox")
+	c.Assert(err, check.NotNil, check.Commentf(out))
+	c.Assert(out, checker.Contains, "invalid value", check.Commentf("couldn't find the invalid value warning"))
+
+	//out, _, err = dockerCmdWithError("search", "--filter", "has-stars=-1", "busybox")
+	//c.Assert(err, check.NotNil, check.Commentf(out))
+	//c.Assert(out, checker.Contains, "invalid value", check.Commentf("couldn't find the invalid value warning"))
```

Comment:
I think we should retain this behavior--I assume the test is commented out because the filter variant of "`--stars`" is not really processed as a flag, whereas `--stars` is defined as a "`Uint`", and therefore the flag processing kicks it out as an invalid value.  If we want to continue calling "-1" invalid, we will have to handle that in our own processing of filter options unless other maintainers feel we don't need to continue with < 0 values as invalid?